lib/edit-site-export.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * REST endpoint for exporting the contents of the Edit Site Page editor.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Output a ZIP file with an export of the current templates
+ * and template parts from the site editor, and close the connection.
+ */
+function edit_site_export() {
+	// Create ZIP file and directories.
+	$filename = tempnam( get_temp_dir(), 'edit-site-export' );
+	$zip      = new ZipArchive();
+	$zip->open( $filename, ZipArchive::OVERWRITE );
+	$zip->addEmptyDir( 'theme' );
```

Comment:
Why not put `block-templates` and `block-template-parts` at the top-level? (Unless I'm wrong about typical theme .zip files having these directories on the top-level)

Reply:
Because unzipping a zip file yields its contents, not its contents wrapped in an extra directory. Try downloading the file to see what I mean.

Reply:
Let's maybe make a constant for the `theme` directory name? We might later want to add an option for the user to give a name to the theme :thinking:

Reply:
We can change it then. I don't see much value in complicating this very simple code with lots of string joins when it's not needed yet.